docs/general/metrics.md

```diff
@@ -89,6 +90,28 @@ When introducing a new metric name check all existing schema files to make sure
 the name does not appear as a key of any "rename_metrics" section (keys denote
 old metric names in rename operations).
 
+### Metric attributes
+
+Metric attributes MUST follow the general [attribute naming rules](attribute-naming.md).
+In particular, metric attributes MUST have a namespace.
+
+Metric attributes SHOULD be added under the metric namespace when their usage and
+semantics are exclusive to the metric.
+
+Examples:
+
+Metric `system.filesystem.usage` with attributes `mode` and `mountpoint`
+should be namespace as: `system.filesystem.mode` and `system.filesystem.mountpoint`.
+
+When metric attributes apply to other signals, these do not need
+to be namespace under the metric and SHOULD be added to the
+[attributes registry](../attributes-registry/README.md).
```

Comment:
> SHOULD be added to the attribute registry

All attributes should be added to the registry, whether or not they are under the metric namespace, right?

Reply:
yes, I'd be in favor of removing this paragraph completely and adding one to https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/attribute-naming.md#recommendations-for-opentelemetry-authors to recommend all attributes to be in the registry.

Reply:
I was a bit torn by this. If we add an attribute that is highly specific to a metric and there's no usage elsewhere, why would we want that to be on the registry? 🤔

Reply:
I think we wanted everything to be in the registry, but that's not normative

> Attributes should be included in this registry before they are used in semantic conventions.

https://github.com/open-telemetry/semantic-conventions/blob/58c2705b7c93ea529d11db991e7ae4ad175473e1/docs/attributes-registry/README.md?plain=1#L15

From the yaml and codegen perspective there is no difference between attributes added to the registry and not added to one - the concept of registry is only captured in the folder structure currently, so I'm taking my comment back (as out of scope of this PR).

Reply:
From my perspective, `discoverability` is one reason and specifically for future contributors. This means that if any future possible usages show up then we won't have to handle duplicates or the need to move to the registry then.

Reply:
Yes, discoverability is the only thing that comes to mind. But again, if it's an attribute that is very particular/specific to a single metric, would it still make sense to be in the registry? Given that maybe we will have a metric registry at some point.. I wonder..

I will add this to the meeting agenda today to see what others think about it.

Reply:
Ok discussed in the semconv meeting on 2024-03-04 and we DO want all attributes in the registry, regardless of their use in a single metric. I will change the wording here.

Reply:
@jack-berg @lmolkova I changed the wording, could you please take a look? Thanks! [e8c31c0](https://github.com/open-telemetry/semantic-conventions/pull/786/commits/e8c31c011c0a4c90651d86494b2b30f5f4b60d10)